javascript/ql/src/experimental/Security/CWE-090/Ldapjs.qll

```diff
@@ -0,0 +1,69 @@
+/**
+ * Provides classes for working with [ldapjs](https://github.com/ldapjs/node-ldapjs) (Client only)
+ */
+
+import javascript
+
+module Ldapjs {
+  string getLdapjsClientDNMethodName() {
+    result = ["add", "bind", "compare", "del", "modify", "modifyDN", "search"]
+  }
+
+  /**
+   * Gets a data flow source node for an LDAP client.
+   */
+  abstract class LdapClient extends DataFlow::SourceNode { }
+
+  /**
+   * Gets a data flow source node for the ldapjs library.
+   */
+  private DataFlow::SourceNode ldapjs() { result = DataFlow::moduleImport("ldapjs") }
+
+  /**
+   * Gets a data flow source node for the ldapjs client.
+   */
+  class LdapjsClient extends LdapClient {
+    LdapjsClient() { this = ldapjs().getAMemberCall("createClient") }
+  }
+
+  /**
+   * Gets a data flow node for the client `search` options.
+   */
+  class LdapjsSearchOptions extends DataFlow::SourceNode {
+    LdapjsSearchOptions() {
+      this = any(LdapClient client).getAMemberCall("search").getArgument(1).getALocalSource()
+    }
+  }
```

Comment:
```suggestion
  class LdapjsSearchOptions extends DataFlow::SourceNode {
    DataFlow::CallNode queryCall;

    LdapjsSearchOptions() {
      queryCall = any(LdapjsClient client).getAMemberCall("search") and
      this = queryCall.getArgument(1).getALocalSource()
    }

    /**
     * Gets the LDAP query call that these options are used in.
     */
    DataFlow::InvokeNode getQueryCall() { result = queryCall }
  }```

Reply:
Thanks for the suggestions. I accepted the previous suggestion (changing `LdapClient` to `LdapjsClient`), but this one I made it manually (it shows outdated version because `LdapClient` is updated)